docs/alerts.md

```diff
@@ -0,0 +1,312 @@
+
+#### Additional Notes
+
+As can be seen, the `notify()` method is simply a placeholder to allow notification handlers to be abstracted from the notification itself.
+This lends itself better to code generators because the `notify()` method can be customized using extern properties so that the act of checking and notifying can be separated from eachother.
+
+## Integration Methods
+
+The above constructs lend themselves well to various different methods of integration and also allow for building out future integration plans.
+
+### Inline
+
+It is possible for the entire query to run inline.
+
+```
+// Send to multiple notification handlers.
+notify = (tables=<-) => {
+	tables |> pagerduty.notify(routingKey: myRoutingKey)
+	tables |> slack.notify(channel: "#flux")
+}
+
+from(bucket: "telegraf")
+    |> range(start: -5m)
+    |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
+    |> check(
+    	crit: (r) => r._value > 90,
+    )
+    |> notify()
+```
+
+### Using Buckets
+
+Using the same script as the above, the `notify` method can be customized to write to a bucket instead and the notifications can be triggered off of the contents of the bucket instead.
+
+```
+// This goes in the check script.
+notify = (tables=<-) => tables |> to(bucket: "system")
+
+// This goes in a separate script.
+alert = (tables=<-, handler) => tables |> handler() |> to(bucket: "notifications")
+
+tables = from(bucket: "system")
+	|> range(start: -1m)
+	|> notify()
+	
+// The function is easy for the user to write and it is easy to
+// inject into a generated script.
+handler1 = (tables=<-) => pagerduty.notify(routingKey: myRoutingKey)
+handler2 = (tables=<-) => slack.notify(channel: "#flux")
+	
+// Send to multiple notification handlers.
+// This is easy to generate.
+tables |> handler1() |> to(bucket: "notifications")
+tables |> handler2() |> to(bucket: "notifications")
+```
+
+### Tasks
+
+Using the above, this can be integrated with tasks fairly easily and meet the above requirements.
+
+This might be the check script.
+
+```
+// Injected.
+option task = {
+	every: 1m,
+}
+
+// Possibly an option like now?
+notify = (tables=<-) => tables |> to(bucket: "system")
+
+// User provided.
+from(bucket: "telegraf")
+    |> range(start: -task.every)
+    |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
+    |> check(
+	crit: (r) => r._value > 90,
+    )
+    |> notify()
+```
+
+This might be the notification script.
+
+```
+// Injected.
+option task = {
+	every: 1m,
+}
+
+tables = from(bucket: "system")
+	|> range(start: -task.every)
+	|> notify()
+
+// The user could customize the function body for each handler,
+// but the function signature would probably be generated.
+handler1 = (tables=<-) => pagerduty.notify(routingKey: myRoutingKey)
+handler2 = (tables=<-) => slack.notify(channel: "#flux")
```

Comment:
>When it comes to the mixins, I was thinking it would be similar to how execute.Aggregate works. I don't know the specifics, but it would be a common implementation that would add additional argument parameters similar to how execute.Aggregate works.

Independent of how the mixins are implemented the Flux script has to declare the  notification rule behavior somewhere. Where is that logic represented in this Flux script? Meaning reading this script how can I tell it is the `notify at most once during a specific interval` behavior vs another behavior.

Reply:
I'll update it with an example.